### PR TITLE
Feature/181

### PR DESCRIPTION
**Pull request by telsaiori**
#181

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class ProjectsController < BaseController
   before_action :find_project, except: [:edit_collection]
 
   def index
-    @projects = current_user.projects
+    @projects = current_user.project_users.is_not_archived.map { |project| Project.find(project.project_id) }
```
Comment (marsz) [inline app/controllers/projects_controller.rb]: 1. I like naming `unarchived` to replace `is_not_archived `
2. I think u need to call one more scope to sort it. Such as `current_user.project_users.unarchived.sorted`
3. `current_user.project_users.map(&:project)` can represent you `map` code block.

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -49,6 +49,24 @@ def destroy
     end
   end
 
+  def archive
+    archive_project = @project.project_users.find_by(user_id: current_user)
```
Comment (marsz) [inline app/controllers/projects_controller.rb]: should be `@project.project_users.find_by(user_id: current_user.id)` ?

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -49,6 +49,24 @@ def destroy
...
+    archive_project.archived = true
+    archive_project.save
```
Comment (marsz) [inline app/controllers/projects_controller.rb]: `archive_project.update(archived: true)` to replace above 2 lines

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -49,6 +49,24 @@ def destroy
...
+    redirect_as_success(edit_projects_path, "project archived")
+  end
+
+  def archived
+    @projects = current_user.project_users.is_archived.map { |project| Project.find(project.project_id) }
```
Comment (marsz) [inline app/controllers/projects_controller.rb]: I like naming `archived` to replace `is_archived`
also map code block as mentioned

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -19,6 +19,10 @@ def collection_for_user_projects(user)
     user.projects.map { |project| [project.name, project.id] }
   end
 
+  def collection_for_not_archived_user_projects(user)
+    user.project_users.is_not_archived.map { |project| Project.find(project.project_id) }
```
Comment (marsz) [inline app/helpers/application_helper.rb]: this collection also should be sorted

File: app/views/layouts/_header.html.slim [hunk 6]
```diff
@@ -9,12 +9,14 @@ nav.navbar.navbar-inverse.navbar-fixed-top
...
         li class=("active" if params[:controller] == "projects" && params[:action] == "new")
           = link_to "新增專案", new_project_path, class: "glyphicon glyphicon-plus"
-        li class=("active" if params[:controller] == "projects" && params[:action] != "new")
+        li class=("active" if params[:controller] == "projects" && params[:action] != "new" && params[:action] != "archived")
           = link_to "我的專案列表", projects_path, class: "glyphicon glyphicon-list-alt"
         li class=("active" if params[:controller] == "records")
           = link_to "我的工時記錄", records_path, class: "glyphicon glyphicon-time"
         li class=("active" if params[:controller] == "todos")
           = link_to "我的 TODO", todos_path, class: "glyphicon glyphicon-th-list"
+        li class=("active" if params[:controller] == "projects" && params[:action] == "archived")
```
Comment (marsz) [inline app/views/layouts/_header.html.slim]: should class name be `archived`?

**Comment by marsz** [comment]
@telsaiori 有一些沒 comment 到的地方可能也會需要修改，感謝

**Comment by telsaiori** [comment]
我更正了，麻煩再幫忙確認一次，謝謝。

**Comment by marsz** [comment]
@telsaiori 感謝 :D

**Pull request merged**